parquet-column/src/test/java/org/apache/parquet/column/values/dictionary/IntListTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.dictionary;
+
+import org.junit.Test;
+
+import junit.framework.Assert;
+
+public class IntListTest {
+
+  /**
+   * Test IntList of fairly small size (< INITIAL_SLAB_SIZE), this tests a single slab being created
+   */
+  @Test
+  public void testSmallList() {
+    int testSize = IntList.INITIAL_SLAB_SIZE - 100;
+    doTestIntList(testSize);
+  }
+
+  /**
+   * Test IntList > INITIAL_SLAB_SIZE so that we have multiple slabs being created
+   */
+  @Test
+  public void testListGreaterThanInitialSlabSize() {
+    int testSize = IntList.INITIAL_SLAB_SIZE + 100;
+    doTestIntList(testSize);
+  }
+
+  /**
+   * Test IntList of a fairly large size (> MAX_SLAB_SIZE) so that we have multiple slabs
+   * created of varying sizes
+   */
+  @Test
+  public void testListGreaterThanMaxSlabSize() {
+    int testSize = IntList.MAX_SLAB_SIZE * 4 + 100;
+    doTestIntList(testSize);
+  }
+
+  private void doTestIntList(int testSize) {
```

Comment:
I think this should also make assertions about the allocated slabs. These tests would have passed before this commit, wouldn't they?